todo/todo.go

```diff
@@ -1,51 +1,42 @@
 package todo
 
 import (
-	"database/sql"
 	"fmt"
+	"practice/go-todo-list/db"
 
 	_ "github.com/lib/pq"
 )
 
-const (
-	host     = "localhost"
-	port     = 5432
-	user     = "postgres"
-	password = ""
-	dbname   = "todo_db"
-)
-
-func dbConn(user, password, dbname string) *sql.DB {
-	dbinfo := fmt.Sprintf("user=%s password=%s dbname=%s sslmode=disable",
-		user, password, dbname)
-	db, err := sql.Open("postgres", dbinfo)
-	if err != nil {
-		fmt.Println("Encountered error: ", err)
-		panic(err)
-	}
-	return db
-}
-
 func Create(task string) (int, error) {
-	db := dbConn(user, password, dbname)
+	db := db.InitDB()
+	// db := dbConn(user, password, dbname)
 	defer db.Close()
 
-	var task_id int
-	statement, err := db.Prepare("INSERT INTO todo_db(task) VALUES($1);")
+	var task_id, dummyVar, mxid int
+
+	// add some code to count total number of rows in db
+	s, err := db.Prepare("SELECT MAX(task_id) AS mxid FROM todo_table;")
+	row := s.QueryRow()
+	row.Scan(&mxid)
+
+	statement, err := db.Prepare("INSERT INTO todo_table(task_id, task, timestamp) VALUES($1, $2, $3);")
 	if err != nil {
 		fmt.Println("Encountered Error: ", err)
 		return 0, err
 	}
-	row := statement.QueryRow(task)
-	row.Scan(&task_id)
-	return task_id, err
+
+	task_id = mxid
+	rows := statement.QueryRow(task_id+1, task, "2018-06-18")
```

Comment:
Since you are using `SERIAL` for the column `task_id`, you need not give it in the query. It is autogenerated. Read [this](http://www.postgresqltutorial.com/postgresql-serial/)

Reply:
Yeah. We were facing troubles with this earlier. Now fixed.